src/main/java/Parser.java

```diff
@@ -0,0 +1,285 @@
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+public class Parser {
+
+
+    public Parser() {
+
+    }
+
+    /**
+     * Parses the String user input from the Ui
+     * Then checks the kind of Task: todo, event, deadline
+     * and also the commands: list, bye, delete, done, find
+     *
+     * @param input string user input from Ui
+     * @throws DukeException  If description is empty, or if the input is none of the commands
+     * @throws IOException If file cannot be written to storage
+     */
+
+    public String parse(String input) {
```

Comment:
I would say this method is too long. The empty lines in the method also make it harder to read. You might consider doing some abstraction!